src/cmd/switch.go

```diff
@@ -14,25 +15,42 @@ import (
 	"github.com/spf13/cobra"
 )
 
-const switchDesc = "Displays the local branches visually and allows switching between them"
+const (
+	switchDesc        = "Displays the local branches visually and allows switching between them"
+	mergeMessageLong  = "merge"
+	mergeMessageShort = "m"
+	mergeDesc         = `When switching branches, if you have local modifications to one or more files that are different between the current branch and the branch to which you are switching, the command refuses to switch branches in order to preserve your modifications in context. However,
+with this option, a three-way merge between the current branch, your working tree contents, and the new branch is done, and you will be on the new branch.`
+)
 
 func switchCmd() *cobra.Command {
 	addVerboseFlag, readVerboseFlag := flags.Verbose()
+	addMergeFlag := func(cmd *cobra.Command) {
+		cmd.PersistentFlags().BoolP(mergeMessageLong, mergeMessageShort, false, mergeDesc)
+	}
+	readMergeFlag := func(cmd *cobra.Command) bool {
+		value, err := cmd.Flags().GetBool("merge")
+		if err != nil {
+			panic(fmt.Sprintf("command %q does not have a bool %q flag", cmd.Name(), mergeMessageLong))
+		}
+		return value
+	}
 	cmd := cobra.Command{
 		Use:     "switch",
 		GroupID: "basic",
-		Args:    cobra.NoArgs,
+		Args:    cobra.MaximumNArgs(2),
```

Comment:
I think this setting is for positional arguments only. Can you please try reverting to `NoArgs`? The flag you added should still work.

Background: [git town compress](https://github.com/git-town/git-town/blob/main/src/cmd/compress.go#L45) uses plenty of CLI flags as well and also has `NoArgs`.

Reply:
NoArgs is defined like this:

`// NoArgs returns an error if any args are included.
func NoArgs(cmd *Command, args []string) error {
	if len(args) > 0 {
		return fmt.Errorf("unknown command %q for %q", args[0], cmd.CommandPath())
	}
	return nil
}
`

I believe ArbitraryArgs is what is required here